seed/ts-sdk/simple-api/use-yarn/src/auth/BearerAuthProvider.ts

```diff
@@ -0,0 +1,33 @@
+// This file was auto-generated by Fern from our API Definition.
+
+import * as core from "../core/index.js";
+import * as errors from "../errors/index.js";
+
+export namespace BearerAuthProvider {
+    export interface Options {
+        token: core.Supplier<string>;
+    }
+}
+
+export class BearerAuthProvider implements core.AuthProvider {
+    private readonly token: core.Supplier<string>;
+
+    constructor(options: BearerAuthProvider.Options) {
+        this.token = options.token;
+    }
+
+    public async getAuthRequest(): Promise<core.AuthRequest> {
+        const token = await core.Supplier.get(this.token);
+        if (token == null) {
+            throw new errors.SeedSimpleApiError({
+                message: "Please specify a token by passing it in to the constructor",
+            });
+        }
+
+        const authHeader = core.BearerToken.toAuthorizationHeader(token);
```

Comment:
Update `toAuthorizationHeader` so that the first parameter is generic, so it can be inferred whether `token` can be undefined, if not undefined, the return should be typed as not undefined too